app/models/partnership_request.rb

```diff
@@ -0,0 +1,52 @@
+# frozen_string_literal: true
+
+class PartnershipRequest < BasePartnership
+  after_create :schedule
+
+  def finalise!
+    ActiveRecord::Base.transaction do
+      partnership = Partnership.create!(
+        delivery_partner: delivery_partner,
+        lead_provider: lead_provider,
+        school: school,
+        cohort: cohort,
+        challenge_deadline: Time.zone.now,
+      )
+      school_cohort = SchoolCohort.find_by!(school_id: school_id, cohort_id: cohort_id)
+      school_cohort.update!(induction_programme_choice: "full_induction_programme", core_induction_programme: nil)
+
+      update_notification_emails(partnership)
+      destroy!
+    end
```

Comment:
Question one: Why do we change challenge deadline? 

Question two, much much bigger: Why do we need a separate model to represent a partnership request? 
Do we expect multiple lead providers to request to partner with one school? If so do we want to destroy all other requests when we finalise this one? 
Maybe a less complex (polymorphic) solution would be to create the partnership when LP requests to partner, and have a status on it that tells us whether it was accepted, rejected or pending? Then you can accomplish all the changes here with a status change.